File: web/src/main/webapp/javascript/work/board/board.js
/**
 * 
 */

 	
	function check(){
		
		var writer = document.getElementById("writer");
		var passwd = document.getElementById("passwd");
		var subject = document.getElementById("subject");
		var content = document.getElementById("content");

		if(writer.value == ""){
			alert("작성자명을 입력");
			writer.focus();
			return false;
		}
		if(passwd.value == ""){
			alert("비밀번호 입력");
			passwd.focus();
			return false;
		}
		if(passwd.value.length < 2 || passwd.value.length > 8){
			alert("비밀번호는 2~8자 이내로 입력");
			passwd.value = "";
			passwd.focus();
			return false;
		}
		if(subject.value == ""){
			alert("제목을 입력");
			subject.focus();
			return false;
		}
		if(subject.value.length > 50){
			alert("제목을 50자 이내로 입력");
			subject.value = "";
			subject.focus();
			return false;
		}
		if(content.value == ""){
			alert("내용을 입력");
			content.focus();
			return false;
		}
		if(content.value.length > 200){
			alert("내용은 200자 이내로 작성");
			content.focus();
			return false;
		}
		
	}

File: part5_Spring_1030/jpaoracle02/src/main/java/com/example/demo/domain/Board.java
package com.example.demo.domain;

import java.sql.Timestamp;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import lombok.Data;

@Data
@Entity
@Table(name="boards")									// boards 테이블 생성
@SequenceGenerator(name="boards_seq_gen", 			    // 시퀀스 제너레이터 이름
 sequenceName="boards_seq", 							// 시퀀스 이름
 initialValue=1, 										// 시작값
 allocationSize=1 )									    // 증가값	
public class Board {
	
	@Id													// 기본키 설정
//	@GeneratedValue(strategy = GenerationType.AUTO)     // JPA 구현체(Hibernate)가 자동으로 생성 방식을 결정
	                                                    // hibernate_sequence가 자동으로 생성된다.
	@GeneratedValue(strategy = GenerationType.SEQUENCE, // 사용할 전략을 시퀀스로 선택
    				generator="boards_seq_gen" )		// 생성기를 boards_seq_gen으로 설정 
	private int no;	
	private String writer;
	private String passwd;
	private String subject;
	private String content;	
	
	@CreationTimestamp
	private Timestamp regdate;	
	
	@UpdateTimestamp
	private Timestamp updatedate;
}
